backends/cadence/hifi/operators/op_div.cpp

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/kernels/portable/cpu/scalar_utils.h>
+#include <executorch/kernels/portable/cpu/util/broadcast_util.h>
+#include <executorch/kernels/portable/cpu/util/functional_util.h>
+#include <executorch/kernels/portable/cpu/util/math_util.h>
+#include <executorch/runtime/kernel/kernel_includes.h>
+#include <executorch/runtime/platform/assert.h>
+#include <cmath> 
+
+#include "kernels.h"
+
+namespace torch {
+namespace executor {
+namespace native {
+    
+#define NNLIB_MAX_DIM 4  /* Add fallback if broadcast and dim > 4 */
+
+namespace {
+
+ScalarType get_compute_type(ScalarType a_type, ScalarType b_type) {
+  if (isFloatingType(a_type) && isFloatingType(b_type)) {
+    return promoteTypes(a_type, b_type);
+  } else if (isFloatingType(a_type)) {
+    return a_type;
+  } else if (isFloatingType(b_type)) {
+    return b_type;
+  }
+  return ScalarType::Float;
+}
+
+} // namespace
+
+Tensor&
+div_out(RuntimeContext& ctx, const Tensor& a, const Tensor& b, Tensor& out) {
+  ET_KERNEL_CHECK(
+      ctx,
+      resize_to_broadcast_target_size(a, b, out) == Error::Ok,
+      InvalidArgument,
+      out);
+
+  ScalarType a_type = a.scalar_type();
+  ScalarType b_type = b.scalar_type();
+
+  ET_KERNEL_CHECK(
+      ctx,
+      !isComplexType(a_type) && !isQIntType(a_type) && !isBitsType(a_type),
+      InvalidArgument,
+      out);
+  ET_KERNEL_CHECK(
+      ctx,
+      !isComplexType(b_type) && !isQIntType(b_type) && !isBitsType(b_type),
+      InvalidArgument,
+      out);
+
+  ET_KERNEL_CHECK(ctx, tensor_is_real_type(out), InvalidArgument, out);
+  
+  int a_dim = a.dim(), b_dim = b.dim(), out_dim = out.dim();
+  int fall_back = 0;
+  /*find broadcast*/
+  const int a_is_broadcasted = !out.sizes().equals(a.sizes());
+  const int b_is_broadcasted = !out.sizes().equals(b.sizes());
+  const int broadcast = (a_is_broadcasted || b_is_broadcasted);
+  int max_dim = a.dim() > b.dim() ? a.dim() : b.dim();
+  max_dim = out.dim() > max_dim ? out.dim() : max_dim;
+  
+  if((a_type != ScalarType::Float) || (b_type != ScalarType::Float))
+    fall_back = 1;
+  
+  if( (a_dim == 0) || (b_dim == 0) )
+    fall_back = 1;
+
+  if((broadcast == 1) && (max_dim > NNLIB_MAX_DIM))
+    fall_back = 1;
+  
+  if(!fall_back)
+  {
```

Comment:
Nit: Can we rewrite this as 
if (optimized) {
  optimized_version(...);
  return;
}
non_optimized_version(...);

Reply:
Okay, will update

Reply:
Done